packages/client/src/components/TransferToken/components/TransactionDetails.js

```diff
@@ -1,24 +1,54 @@
 import { useContext, useEffect, useState } from 'react';
 import { Web3Context } from 'contexts/Web3';
-import { useClipboard, useFlowFees } from 'hooks';
-import { ASSET_TYPES } from 'constants/enums';
+import { useClipboard, useContacts, useFlowFees } from 'hooks';
+import { ASSET_TYPES, TRANSACTION_TYPE } from 'constants/enums';
 import { COIN_TYPE_TO_META } from 'constants/maps';
+import { getNameByAddress } from 'utils';
 import Svg from 'library/Svg';
 import { isEmpty } from 'lodash';
-import { SendTokensContext } from '../sendTokensContext';
+import { TransferTokensContext } from '../TransferTokensContext';
 
 const TransactionDetails = () => {
   const [transactionFee, setTransactionFee] = useState(0);
-  const [sendModalState] = useContext(SendTokensContext);
-  const { assetType, coinType, tokenAmount, selectedNFT, recipient, address } =
-    sendModalState;
-  const web3 = useContext(Web3Context);
+  const [sendModalState] = useContext(TransferTokensContext);
+  const {
+    assetType,
+    coinType,
+    tokenAmount,
+    selectedNFT,
+    recipient,
+    sender,
+    transactionType,
+  } = sendModalState;
 
-  const { safeOwners } = web3?.treasuries?.[address];
-  const recipientName =
-    safeOwners.find((owner) => owner.address === recipient)?.name ?? '';
   const clipboard = useClipboard();
+  const web3 = useContext(Web3Context);
+  const safeAddress =
+    transactionType === TRANSACTION_TYPE.SEND ? sender : recipient;
+
+  const { contacts } = useContacts(safeAddress);
+  const safeName = web3?.treasuries?.[safeAddress].name;
+
+  let senderName;
+  let recipientName;
+  let label;
+  switch (transactionType) {
+    case TRANSACTION_TYPE.SEND:
+      senderName = safeName;
+      recipientName = getNameByAddress(contacts, recipient);
+      label = 'Sent';
+      break;
+    case TRANSACTION_TYPE.DEPOSIT:
+      senderName = getNameByAddress(contacts, sender);
+      recipientName = safeName;
+      label = 'Deposit';
+      break;
+    default:
+      break;
+  }
```

Comment:
since theres only 2 conditions this would probably be clearer as an if/else statement - unless there is other transaction types coming in which case switch is probably better

Reply:
No, it's only two of them, I'll update it!